sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -312,13 +321,18 @@ private static void CloseLink(RequestResponseAmqpLink link)
                         {
                             link.DisposeDelivery(message, true, AmqpConstants.AcceptedOutcome);
                         }
+
                         receivedMessages.Add(AmqpMessageConverter.AmqpMessageToSBMessage(message));
                         message.Dispose();
                     }
                 }
 
                 return receivedMessages;
             }
+            catch (OperationCanceledException) when(cancellationToken.IsCancellationRequested)
```

Comment:
```suggestion
            catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)
```

Reply:
Yeah sorry it was really just a sloppy spike to see what you think about this approach

Reply:
All good.  I just can't turn off the OCD when reading things.  😛